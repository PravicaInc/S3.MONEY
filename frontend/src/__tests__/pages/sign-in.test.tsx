import { cleanup, render, RenderResult } from '@testing-library/react';

import '@testing-library/jest-dom';

import SignInPage from '@/app/sign-in/page';

import {
  DEFAULT_WALLET_WITH_CORRECT_STATUS_VALUE,
  WalletWithCorrectStatusContext,
  WalletWithCorrectStatusContextProps,
} from '@/hooks/useWallet';

const renderSignInPageWithWalletProps = (
  { walletProps, ...renderOptions }: { walletProps: Partial<WalletWithCorrectStatusContextProps> }
) => render(
  <WalletWithCorrectStatusContext.Provider
    value={{
      ...DEFAULT_WALLET_WITH_CORRECT_STATUS_VALUE,
      ...walletProps,
    }}
  >
    <SignInPage />
  </WalletWithCorrectStatusContext.Provider>,
  renderOptions
);

describe('Sign-in page:', () => {
  afterAll(cleanup);

  describe('Sign-in page with loading state:', () => {
    let renderResult: RenderResult;

    beforeEach(() => {
      renderResult = renderSignInPageWithWalletProps(
        {
          walletProps: {
            status: 'connecting',
            connecting: true,
            connected: false,
            disconnected: false,
          },
        }
      );
    });

    afterEach(() => renderResult.unmount);

    it('Show loader', () => {
      expect(renderResult.queryByTestId('loader')).toBeVisible();
    });
  });

  describe('Sign-in page with disconnected state:', () => {
    let renderResult: RenderResult;

    beforeEach(() => {
      renderResult = renderSignInPageWithWalletProps(
        {
          walletProps: {
            status: 'disconnected',
            connecting: false,
            connected: false,
            disconnected: true,
          },
        }
      );
    });

    afterEach(() => renderResult.unmount);

    it('Don`t show loader', () => {
      expect(renderResult.queryByTestId('loader')).toBeNull();
    });

    it('Show wallet connect button', () => {
      expect(renderResult.queryByText('Sign In with Sui Wallet')).toBeVisible();
    });

    it('Snapshot', () => {
      expect(renderResult.container).toMatchSnapshot();
    });
  });

  describe('Sign-in page with connected state:', () => {
    let renderResult: RenderResult;
    const shortWalletAddress = '0x11...1111';

    beforeEach(() => {
      renderResult = renderSignInPageWithWalletProps(
        {
          walletProps: {
            shortWalletAddress,
            status: 'connected',
            connecting: false,
            connected: true,
            disconnected: false,
          },
        }
      );
    });

    afterEach(() => renderResult.unmount);

    it('Show redirect', () => {
      expect(renderResult.queryByText('Redirecting ...')).toBeVisible();
    });
  });
});
