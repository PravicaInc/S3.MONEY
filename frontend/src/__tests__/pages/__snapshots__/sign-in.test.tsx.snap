// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sign-in page: Sign-in page with disconnected state: Snapshot 1`] = `
<div>
  <div
    class="flex flex-col h-screen"
  >
    <div
      class="grid md:grid-cols-2 h-full"
    >
      <div
        class="bg-white h-full flex items-center justify-center py-5"
      >
        <div
          class="flex flex-col items-center mx-20 max-w-4xl text-center"
        >
          <svg />
          <p
            class="mt-4 text-primary text-[32px] leading-[40px]"
          >
            StableCoin Studio on Sui
          </p>
          <div
            class="mt-5 w-full"
          >
            <div
              aria-controls="radix-:r6:"
              aria-expanded="false"
              aria-haspopup="dialog"
              data-state="closed"
              type="button"
            >
              <button
                class="wkit-button bg-actionPrimary text-white p-4 rounded-xl shadow-button border-actionSecondary border-solid border font-semibold hover:bg-white hover:text-actionPrimary transition disabled:bg-slate-300 disabled:border-slate-400 disabled:text-white py-[14px] w-full"
              >
                Sign In with Sui Wallet
              </button>
            </div>
          </div>
          <p
            class="mt-5 text-primary lett"
          >
            Donâ€™t have Sui Wallet extension installed?
             
            <a
              class="text-actionPrimary font-semibold"
              href="https://chromewebstore.google.com/detail/sui-wallet/opcgpfmipidbgpenhmajoajpbobppdil"
              rel="noreferrer"
              target="_blank"
            >
              Install Extension
            </a>
          </p>
        </div>
      </div>
      <div
        class="bg-[#FFCDA8] h-full border-[10px] border-white flex items-center justify-center py-5"
      >
        <div
          class="text-center mx-5 md:mx-10 lg:mx-20 max-w-4xl"
        >
          <p
            class="text-[40px] leading-[44px] font-semibold text-primary"
          >
            Gateway to Code-Free StableCoin Creation
          </p>
          <p
            class="mt-5 text-[#353849] text-opacity-70"
          >
            Just connect your Sui wallet and step into the wonderfully simple world of StableCoin Studio on the Sui Blockchain. You'll be able to create and manage your own stable coins with just a few clicks. And the best part? No coding is required! It's a platform designed just for you.
          </p>
          <img
            alt="Sui wallet login screen"
            class="mx-auto mt-14"
            data-nimg="1"
            decoding="async"
            height="525"
            loading="lazy"
            src="/_next/image?url=%2Fimages%2Fsui_wallet_login_screen.png&w=640&q=75"
            srcset="/_next/image?url=%2Fimages%2Fsui_wallet_login_screen.png&w=384&q=75 1x, /_next/image?url=%2Fimages%2Fsui_wallet_login_screen.png&w=640&q=75 2x"
            style="color: transparent;"
            width="290"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
